test/ambiguous.jl

```diff
@@ -133,7 +133,8 @@ ambs = detect_ambiguities(Ambig5)
 @test length(ambs) == 2
 
 # Test that Core and Base are free of ambiguities
-@test detect_ambiguities(Core, Base; imported=true) == []
+# TODO jb/subtype: we now detect a lot more
+@test_broken detect_ambiguities(Core, Base; imported=true) == []
```

Comment:
tkelman is correct. By monkey-patching the ambiguity test to remove any signatures that set a static parameter to Bottom, the list is fairly short:
```julia
julia> eval(Base, quote
function isambiguous(m1::Method, m2::Method)
    ti = typeintersect(m1.sig, m2.sig)
    isbottom(x::ANY) = (x === Bottom)
    isbottom(ti) && return false
    (ti1, env1) = ccall(:jl_match_method, Any, (Any, Any), ti, m2.sig)::SimpleVector
    any(isbottom, env1) && return false
    (ti2, env2) = ccall(:jl_match_method, Any, (Any, Any), ti, m2.sig)::SimpleVector
    any(isbottom, env2) && return false
    ml = _methods_by_ftype(ti, -1, typemax(UInt))
    isempty(ml) && return true
    for m in ml
        if ti <: m[3].sig
            return false
        end
    end
    return true
end
end)

julia> for a in Test.detect_ambiguities(Base, imported=true)
         println(a[1])
         println(a[2])
         println("=> ", typeintersect(a[1].sig, a[2].sig))
         println()
       end

copy!(dest::Base.PermutedDimsArrays.PermutedDimsArray, src::Base.SparseArrays.CHOLMOD.Dense) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1009
copy!{T<:Union{Complex{Float64},Float64}}(dest::AbstractArray{T,2}, D::Base.SparseArrays.CHOLMOD.Dense{T}) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1012
=> Tuple{Base.#copy!,Base.PermutedDimsArrays.PermutedDimsArray{T,2,perm,iperm,AA} where AA<:AbstractArray where iperm where perm,Base.SparseArrays.CHOLMOD.Dense{T}} where T<:Union{Complex{Float64},Float64}

copy!(dest::Base.PermutedDimsArrays.PermutedDimsArray, src::AbstractArray) in Base.PermutedDimsArrays at permuteddimsarray.jl:100
copy!{T<:Union{Complex{Float64},Float64}}(dest::AbstractArray{T,N} where N, D::Base.SparseArrays.CHOLMOD.Dense{T}) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1011
=> Tuple{Base.#copy!,Base.PermutedDimsArrays.PermutedDimsArray{T,N,perm,iperm,AA} where AA<:AbstractArray where iperm where perm where N,Base.SparseArrays.CHOLMOD.Dense{T}} where T<:Union{Complex{Float64},Float64}

copy!{T,N}(dest::Base.PermutedDimsArrays.PermutedDimsArray{T,N,perm,iperm,AA} where AA<:AbstractArray where iperm where perm, src::AbstractArray{T,N}) in Base.PermutedDimsArrays at permuteddimsarray.jl:97
copy!(dest::AbstractArray, D::Base.SparseArrays.CHOLMOD.Dense) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1013
=> Tuple{Base.#copy!,Base.PermutedDimsArrays.PermutedDimsArray{T,2,perm,iperm,AA} where AA<:AbstractArray where iperm where perm,Base.SparseArrays.CHOLMOD.Dense{T}} where T<:Union{Complex{Float64},Float64}

copy!(dest::Base.PermutedDimsArrays.PermutedDimsArray, src::Base.SparseArrays.CHOLMOD.Dense) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1009
copy!{T<:Union{Complex{Float64},Float64}}(dest::AbstractArray{T,N} where N, D::Base.SparseArrays.CHOLMOD.Dense{T}) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1011
=> Tuple{Base.#copy!,Base.PermutedDimsArrays.PermutedDimsArray{T,N,perm,iperm,AA} where AA<:AbstractArray where iperm where perm where N,Base.SparseArrays.CHOLMOD.Dense{T}} where T<:Union{Complex{Float64},Float64}

\{T<:Union{Float32,Float64}}(F::Factorization{T}, B::Union{Array{Complex{T},1},Array{Complex{T},2}}) in Base.LinAlg at linalg/factorization.jl:36
\{T<:Union{Complex{Float64},Float64}}(L::Base.SparseArrays.CHOLMOD.Factor{T}, B::Union{Base.ReshapedArray{T,2,A,MI} where MI<:Tuple{Vararg{Base.MultiplicativeInverses.SignedMultiplicativeInverse{Int64},N} where N} where A<:DenseArray,DenseArray{T,2},SubArray{T,2,A,I,L} where L} where I<:Tuple{Vararg{Union{Base.AbstractCartesianIndex,Colon,Int64,Range{Int64}},N} where N} where A<:Union{Base.ReshapedArray{T,N,A,MI} where MI<:Tuple{Vararg{Base.MultiplicativeInverses.SignedMultiplicativeInverse{Int64},N} where N} where A<:DenseArray where N where T,DenseArray} where T) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1582
=> Tuple{Base.#\,Base.SparseArrays.CHOLMOD.Factor{T},Array{Complex{T},2}} where T<:Float64

unsafe_convert{T}(P::Type{Ptr{T}}, b::Base.RefArray{T,A,R} where R where A<:AbstractArray) in Base at refpointer.jl:78
unsafe_convert{T}(::Type{Ptr{Void}}, b::Base.RefArray{T,A,R} where R where A<:AbstractArray) in Base at refpointer.jl:87
=> Tuple{Base.#unsafe_convert,Type{Ptr{Void}},Base.RefArray{Void,A,R} where R where A<:AbstractArray}

\{T<:Union{Float32,Float64}}(F::Factorization{T}, B::Union{Array{Complex{T},1},Array{Complex{T},2}}) in Base.LinAlg at linalg/factorization.jl:36
\{T<:Union{Complex{Float64},Float64}}(L::Base.SparseArrays.CHOLMOD.Factor{T}, b::Union{Base.ReshapedArray{T,1,A,MI} where MI<:Tuple{Vararg{Base.MultiplicativeInverses.SignedMultiplicativeInverse{Int64},N} where N} where A<:DenseArray,DenseArray{T,1},SubArray{T,1,A,I,L} where L} where I<:Tuple{Vararg{Union{Base.AbstractCartesianIndex,Colon,Int64,Range{Int64}},N} where N} where A<:Union{Base.ReshapedArray{T,N,A,MI} where MI<:Tuple{Vararg{Base.MultiplicativeInverses.SignedMultiplicativeInverse{Int64},N} where N} where A<:DenseArray where N where T,DenseArray} where T) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1581
=> Tuple{Base.#\,Base.SparseArrays.CHOLMOD.Factor{T},Array{Complex{T},1}} where T<:Float64

copy!{T,N}(dest::Base.PermutedDimsArrays.PermutedDimsArray{T,N,perm,iperm,AA} where AA<:AbstractArray where iperm where perm, src::AbstractArray{T,N}) in Base.PermutedDimsArrays at permuteddimsarray.jl:97
copy!{T<:Union{Complex{Float64},Float64}}(dest::AbstractArray{T,2}, D::Base.SparseArrays.CHOLMOD.Dense{T}) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1012
=> Tuple{Base.#copy!,Base.PermutedDimsArrays.PermutedDimsArray{T,2,perm,iperm,AA} where AA<:AbstractArray where iperm where perm,Base.SparseArrays.CHOLMOD.Dense{T}} where T<:Union{Complex{Float64},Float64}

unsafe_convert{T}(P::Type{Ptr{T}}, b::Base.RefValue{T}) in Base at refpointer.jl:54
unsafe_convert{T}(::Type{Ptr{Void}}, b::Base.RefValue{T}) in Base at refpointer.jl:63
=> Tuple{Base.#unsafe_convert,Type{Ptr{Void}},Base.RefValue{Void}}

copy!(dest::Base.PermutedDimsArrays.PermutedDimsArray, src::AbstractArray) in Base.PermutedDimsArrays at permuteddimsarray.jl:100
copy!{T<:Union{Complex{Float64},Float64}}(dest::AbstractArray{T,2}, D::Base.SparseArrays.CHOLMOD.Dense{T}) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1012
=> Tuple{Base.#copy!,Base.PermutedDimsArrays.PermutedDimsArray{T,2,perm,iperm,AA} where AA<:AbstractArray where iperm where perm,Base.SparseArrays.CHOLMOD.Dense{T}} where T<:Union{Complex{Float64},Float64}

copy!{T,N}(dest::Base.PermutedDimsArrays.PermutedDimsArray{T,N,perm,iperm,AA} where AA<:AbstractArray where iperm where perm, src::AbstractArray{T,N}) in Base.PermutedDimsArrays at permuteddimsarray.jl:97
copy!(dest::Base.PermutedDimsArrays.PermutedDimsArray, src::Base.SparseArrays.CHOLMOD.Dense) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1009
=> Tuple{Base.#copy!,Base.PermutedDimsArrays.PermutedDimsArray{T,2,perm,iperm,AA} where AA<:AbstractArray where iperm where perm,Base.SparseArrays.CHOLMOD.Dense{T}} where T<:Union{Complex{Float64},Float64}

convert{T,N}(::Type{AbstractArray{T,N}}, A::AbstractArray{T,N}) in Base at abstractarray.jl:738
convert{T,N}(::Type{AbstractArray{T,N}}, B::BitArray{N}) in Base at bitarray.jl:594
=> Tuple{Base.#convert,Type{AbstractArray{Bool,N}},BitArray{N}} where N

convert{Tv}(::Type{SparseVector{Tv,Ti} where Ti<:Integer}, s::AbstractArray{Tv,1}) in Base.SparseArrays at sparse/sparsevector.jl:293
convert{Tv,TvS,TiS}(::Type{SparseVector{Tv,Ti} where Ti<:Integer}, s::SparseVector{TvS,TiS}) in Base.SparseArrays at sparse/sparsevector.jl:306
=> Tuple{Base.#convert,Type{SparseVector{Tv,Ti} where Ti<:Integer},SparseVector{Tv,TiS}} where TiS where Tv

convert{Tv,Ti}(::Type{SparseVector{Tv,Ti}}, s::AbstractArray{Tv,1}) in Base.SparseArrays at sparse/sparsevector.jl:290
convert{Tv,Ti,TvS,TiS}(::Type{SparseVector{Tv,Ti}}, s::SparseVector{TvS,TiS}) in Base.SparseArrays at sparse/sparsevector.jl:303
=> Tuple{Base.#convert,Type{SparseVector{Tv,Ti}},SparseVector{Tv,TiS}} where TiS where Ti where Tv

convert{T}(::Type{Factorization{T}}, F::Factorization{T}) in Base.LinAlg at linalg/factorization.jl:30
convert{T,S,U}(::Type{Factorization{T}}, F::Base.LinAlg.LDLt{S,U}) in Base.LinAlg at linalg/ldlt.jl:16
=> Tuple{Base.#convert,Type{Factorization{T}},Base.LinAlg.LDLt{T,U}} where U where T

copy!{T,N}(dest::Base.PermutedDimsArrays.PermutedDimsArray{T,N,perm,iperm,AA} where AA<:AbstractArray where iperm where perm, src::AbstractArray{T,N}) in Base.PermutedDimsArrays at permuteddimsarray.jl:97
copy!{T<:Union{Complex{Float64},Float64}}(dest::AbstractArray{T,N} where N, D::Base.SparseArrays.CHOLMOD.Dense{T}) in Base.SparseArrays.CHOLMOD at sparse/cholmod.jl:1011
=> Tuple{Base.#copy!,Base.PermutedDimsArrays.PermutedDimsArray{T,2,perm,iperm,AA} where AA<:AbstractArray where iperm where perm,Base.SparseArrays.CHOLMOD.Dense{T}} where T<:Union{Complex{Float64},Float64}
```